src/integration_test/java/com/klodnicki/Bike/v2/rest/controller/AdminUserControllerIntegrationTest.java

```diff
@@ -29,25 +31,45 @@ class AdminUserControllerIntegrationTest {
     private UserRepository userRepository;
     @Autowired
     private ModelMapper modelMapper;
+    @Autowired
+    private UserServiceHandler userServiceHandler;
     private User user1;
     private User user2;
+    private String basicAuthHeader;
+    private Authority authority;
+    private Authority authority2;
+    @Autowired
+    private AuthorityRepository authorityRepository;
 
     @BeforeEach
     public void setUp() {
+        String password = "password";
         userRepository.deleteAll();
+        authority = new Authority(null, "ROLE_ADMIN");
+        authority2 = new Authority(null, "ROLE_ADMIN");
+        Set<Authority> authoritySet = new HashSet<>();
+        authoritySet.add(authority);
+
+        Set<Authority> authoritySet2 = new HashSet<>();
+        authoritySet2.add(authority2); //must have separate authority otherwise I get "detached entity passed to persist"
```

Comment:
This detached entity passed to persist when exactly happens? Could you describe it further for me, please?

Reply:
if I create two user instances and to both I provide the same Set<Authority> like here:

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/b9f14e04-34d6-4797-8aac-87249730de09)

then I get this:

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/58d2d53f-7f55-411e-a8e2-17ea271ad830)

Reply:
If I do like this:

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/f4fcce2d-9a47-4792-b448-5a1b197accab)

everything is fine.

I guess it can have something in common that maybe they are being created at once in the same transaction? 
tell me why... 
![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/3325b8b4-adea-4af3-859e-26574f61665b)

Reply:
I think the issue is exactly opposite what you described. In my opinion it happens when you add (save/persist) the first user which have same authorities as the other user. This creates detached entity, because it "points" to user2 which is not yet saved in database.
My guess above could then be resolved if you could try and save authority first and then the users. That could help, but tbh I'm not sure :)
My second guess would be to save user1 and user2 at the same time with this authority – in same transaction – so everything is correctly saved at once. This could prevent detached state.

Reply:
(btw I love last image <3)

Reply:
What I meant was they were being created at once (without saving first user in repo) and then passing this authority set again, this time to the other user (without saving it first).
So we have similar observations. I will make some changes and experiment so I will get more into  its behaviour.
Could I for example set @Transactional to this setUp() or rather it is not what is usaully is being done in integration testing?

Reply:
nope it does NOT work neither this:

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/41ccc9b3-3b4b-4034-8786-41ef45f3f70b)

nor this:

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/60380982-920b-402a-b484-b3d3923b1625)